DUI: Shorten the tool tip description for long texts

aosyatnik opened the pull request:
If description is too long, we use scrollviewer to show it.

Before:
![image](https://cloud.githubusercontent.com/assets/8158404/6263649/21282336-b823-11e4-9bce-20af2279ab5b.png)

After:
![image](https://cloud.githubusercontent.com/assets/8158404/6263654/30ae5988-b823-11e4-8554-b6de525eb2be.png)

Reviewers
@Benglin ,@vmoyseenko, please take a look.

Link to YouTrack:
[MAGN-6211](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-6211) [DUI]Shorten the tool tip description for long texts

Benglin commented on src/DynamoCoreWpf/Views/Search/LibraryView.xaml.cs:
```diff
@@ -57,6 +57,13 @@ private void OnPreviewMouseWheel(object sender, MouseWheelEventArgs e)
             }
```
Hi @aosyatnik, if you could update the comments as follow, that will be great:

```csharp
// Case 1: User scrolls the mouse wheel when Shift key is being held down. This 
// action triggers a horizontal scroll on the library view (so that lengthy names 
// can be revealed). Setting 'Handled' to 'false' allows the underlying scroll bar
// to handle the mouse wheel event.

// Case 2: If the mouse is outside of the library view, but mouse wheel messages 
// get sent to it anyway. In such case there is nothing to change here. The 'Handled'
// is not set to 'true' here because the mouse wheel messages should be routed to the 
// ScrollViewer on tool-tip for further processing.

// Case 3: 
// Mouse wheel without any modifier keys, it scrolls the library view vertically. In 
// this case 'VerticalOffset' is updated, 'Handled' is also set so that mouse wheel 
// message routing ends here.
```

Benglin commented on src/DynamoCoreWpf/Controls/TooltipWindow.xaml:
```diff
@@ -21,6 +21,155 @@
             <controls:BooleanToBrushConverter x:Key="DescriptionToColorConverter"
                                               TrueBrush="#cccccc"
                                               FalseBrush="#666666" />
+
```
I'm just curious, is there a way to get these styles from other places like `DynamoModern.xaml`? If we can, then both `LibrarySearchView.xaml` and `LibraryView.xaml` should be using a centralized one. If we can't use the one in `DynamoModern.xaml`, then maybe at least we can make a new one just to share between both `LibrarySearchView.xaml` and `LibraryView.xaml`. Please let me know if that's possible :)

aosyatnik commented:
Hi, @Benglin . I have moved all parts, those are connected with scrolling to SidebarGridStyleDictionary. xaml. And removed these parts from LibraryView and LibrarySearchView. I tried move other parts to this dictionary, but I couldn't. If I move templates like expanders or buttons, everything works very slow.

Benglin commented on src/DynamoCoreWpf/UI/Themes/Modern/SidebarGridStyleDictionary.xaml:
```diff
@@ -0,0 +1,220 @@
+    <!-- Region LibraryView + LibrarySearchView-->
+
+    <ControlTemplate x:Key="LibraryThumbTemplate"
+                     TargetType="{x:Type Thumb}">
```
Between `LibraryThumbTemplate` and `TooltipThumbTemplate`, only `Width` value is different (one is `5` another is `7`). Hi @elayabharath, can we have the same thumb width for both library view and the tool-tip? If so we can merge these two templates into one.

elayabharath replied:
Yep let's merge them both - 7px for both should be okay.

Benglin commented on src/DynamoCoreWpf/UI/Themes/Modern/SidebarGridStyleDictionary.xaml:
```diff
@@ -0,0 +1,220 @@
+        </Grid>
+    </ControlTemplate>
+
+    <ControlTemplate x:Key="TooltipScrollBarTemplate"
```
`LibraryScrollBarTemplate` and `TooltipScrollBarTemplate` can be merged, if we can merge the above two.

Benglin commented on src/DynamoCoreWpf/UI/Themes/Modern/SidebarGridStyleDictionary.xaml:
```diff
@@ -0,0 +1,220 @@
+        </Grid>
+    </ControlTemplate>
+
+    <ControlTemplate x:Key="LibraryScrollViewerControlTemplate"
```
`LibraryScrollViewerControlTemplate` and `TooltipScrollViewerControlTemplate` are mostly the same, except `Width` and `Style`. I know that `Style` cannot be unified in this case (due to difference in their behaviors), but can we specify the `Style` and `Width` separately in the place where the template is used? Or is that not possible?

Benglin commented:
Thanks for doing this, @aosyatnik. I know XAML extraction can be an extremely confusing process (mostly because XAML contents are hard to read). I have mentioned a few things, if you can further merge them (since @elayabharath has agreed to 7px thumb width), let me know, I'll merge it in.

aosyatnik commented:
Hi, @Benglin , since @elayabharath  is okay with 7 px thumb I can use just 1 Template as well as Style for tooltip and library view. Thanks.

Benglin commented:
Thanks again, @aosyatnik, this is so much better now! LGTM :)
